### DISCUSSION TITLE
access environment inside the expectation expression

### DISCUSSION THREAD

**Discussion by angelabriel** [trento-project/wanda#120] [Q&A] [answered]
The documentation https://github.com/trento-project/wanda/blob/main/guides/specification.md?plain=1#L391 stated:

"the expectation expression itself with access to gathered facts, resolved values and the environment"

How can I access environment variables inside of 'expect:' expressions?
Using the same syntax as in the 'values: conditions' - e.g. env.provider - does not work

```
    expect: |
       if env.provider == "azure" || env.provider == "kvm" {
         facts.sbd_SBD_PACEMAKER == values.expected_SBD_PACEMAKER
       } else {
         true
       }
```

As result I get
`
"expectation_evaluations":[{"message":"Variable not found: env (line 1, position 5)","name":"expectations_SBD_PACEMAKER","type":"variable_not_found"}],
`

**Comment by fabriziosestito** [comment, marked as answer]
@angelabriel ATM is not possible accessing the env from expectations since the env is only used to compute values. Do you still need it after answer #121 ? If not I’ll just open a ticket about correcting the documentation.

**Reply by angelabriel** [reply]
No thanks, for the moment I do not have the need to access the environment from expectations. If I will find another situation, we can restart/continue this discussion with the new use case.
But enhancing the documentation to make this sentence a little bit more clear is highly appreciated, thanks
